### PR TITLE
HACK week: AI Thank you note

### PR DESCRIPTION
**Pull request by hafizrahman**
### Description

Adds a "Thank-you note" generator utilizing Jetpack AI.

This feature is available on Orders with "Completed" status. A generated note can then be copy/pasted, or shared using the Android share sheet.

As this is a HACK week work, the design/wording will likely need further iteration. For now please focus on functionality and code for the review.

Check video below for demo:

[ai-thank-you-note-demo.webm](https://github.com/woocommerce/woocommerce-android/assets/266376/4611d77f-18f7-42e0-8d16-3f5b6afd8616)

### Testing instructions

1. Use a WordPress.com site,
2. Find an Order with "Completed" status, open it,
3. Tap the "Create thank you note" button,
4. Wait and ensure a thank-you note is generated,
5. Try also the copy, feedback, "Regenerate" and "Share" option. (For feedback you will need to check in Logcat whether it's logged properly or not).
6. Also try turning off internet while Regenerating to test error UI.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt [hunk 1]
```diff
@@ -810,6 +817,29 @@ class OrderDetailViewModel @Inject constructor(
         shippingLabelOnboardingRepository.markWcShippingBannerAsDismissed()
     }
 
+    fun onAIThankYouNoteButtonClicked() {
+        launch {
+            val orderRefunds = loadOrderRefunds()
+            val orderProducts = loadOrderProducts(orderRefunds)
+
+            val firstProductId = when (val first = orderProducts.list.first()) {
```
Comment (rossanafmenezes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt:795]: Here, if the user opens an order generated via Simple Payments ( and soon via order creation), meaning no products and no product ID, the app crashes. 

```
 FATAL EXCEPTION: main
                 Process: com.woocommerce.android.dev, PID: 7642
                 java.util.NoSuchElementException: List is empty.
                 	at kotlin.collections.CollectionsKt___CollectionsKt.first(_Collections.kt:214)
                 	at com.woocommerce.android.ui.orders.details.OrderDetailViewModel$onAIThankYouNoteButtonClicked$1.invokeSuspend(OrderDetailViewModel.kt:825)
```

thank  you note

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt [hunk 2]
```diff
@@ -779,7 +783,10 @@ class OrderDetailViewModel @Inject constructor(
             wcShippingBannerVisible = shippingLabelOnboardingRepository.shouldShowWcShippingBanner(
                 order,
                 orderEligibleForInPersonPayments
-            )
+            ),
+            isAIThankYouNoteButtonShown = FeatureFlag.AI_ORDER_DETAIL_THANK_YOU_NOTE.isEnabled() &&
+                selectedSite.get().isWPComAtomic &&
+                order.status == Order.Status.Completed
```
Comment (rossanafmenezes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt]: So, maybe we could hide the button for orders with no products?

Something like:

```
 && productList.value?.isNotEmpty() == true
```
Or maybe if the product list is empty show a snackbar clarifying the flow.

**Review by rossanafmenezes — CHANGES_REQUESTED** [review]
👋  Hi @hafizrahman 

Awesome feature! Looks great. It works as expected. However, I found an issue. The app crashes if the order has no products. Kindly take a look.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt (hunk 1 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt:795]: Thanks for mentioning that! Did not know this is a thing, will fix it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt (hunk 2 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt]: The thank you AI generation now currently requires an existing product to populate the content, so I suppose it's better to hide it :)

Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt]: Resolved in https://github.com/woocommerce/woocommerce-android/pull/9749/commits/7ff1852188bd16f9ab3d1b84d275ee7bcd558eb8

**Review by rossanafmenezes — APPROVED** [review]
Thanks for fixing it @hafizrahman 

LGTM

**Comment by JorgeMucientes** [comment]
Hey @hafizrahman, friendly ping 🙂 . Is there anything preventing this from being merged other than resolving the conflicts?

**Comment by hafizrahman** [comment]
Not sure about the new danger errors "Please add tests for class...". I'm seeing other classes like `AIProductDescriptionViewModel` also don't have tests 🤔 

Since we're close to code freeze, I'm merging this first and will add in the unit tests later. 🙏🏼

**Pull request merged**